File: juezPatito/2319/main.py
import math as m

def verifica(n):
	return m.ceil(m.sqrt(float(n))) == m.floor(m.sqrt(float(n)))

t = int(input())

while t > 0:
	n = int(input())

	x = 1

	for i in list(input().split()):
		x = x * int(i)

	if verifica(x):
		print("EL producto es numero cuadrado")
	else:
		print("El producto no es numero cuadrado")

	t -= 1

File: juezPatito/2319/main.cpp
#include<bits/stdc++.h>

using namespace std;

bool verifica(float n)
{
	return ceil(n) == floor(n);
}

int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int n, t;

	float x, val;

	cin >> n;

	while(n--)
	{
		cin >> t;

		x = 1;

		while(t--)
		{
			cin >> val;
			x = x * sqrt(val);
		}
		/*
		if(verifica(x))
			cout << "EL producto es numero cuadrado\n";
		else
			cout << "El producto no es numero cuadrado\n";
		*/
		cout << to_string(ceil(x)) << " " << to_string(floor(x)) << "\n";
	}
	return 0;
}